### PR TITLE
WP-10970 : fix tap-tqp/tap-s3-csv integration

### PR DESCRIPTION
**Pull request by woody-feng**
https://varicent.atlassian.net/browse/WP-10970

If the metadata contains `source_type` it will output the value as `str(value)`, otherwise output value based on actual schema type. (infered schema or schema by `TQP`)

### CODE REVIEW

File: tap_s3_csv/transform.py [hunk 1]
```diff
@@ -372,4 +384,9 @@ def resolve_filter_fields(metadata=None):
             if (selected is False) or (inclusion == 'unsupported'):
                 filters.add(breadcrumb)
 
-    return frozenset(autos), frozenset(filters)
+            source_type = singer.metadata.get(
+                metadata, breadcrumb, 'source_type')
+            if source_type:
+                source_type_map[breadcrumb_path(breadcrumb)] = source_type
+
+    return frozenset(autos), frozenset(filters), source_type_map
```
Comment (yi-varicent) [inline tap_s3_csv/transform.py:392]: should this be frozenset too

File: tap_s3_csv/transform.py [hunk 2]
```diff
@@ -187,7 +187,7 @@ def _transform_anyof(self, data, schema, path):
         subschemas = schema['anyOf']
         for subschema in subschemas:
             success, transformed_data = self.transform_recur(
-                data, subschema, path)
+                data, subschema, path, )
```
Comment (kchan-varicent) [inline tap_s3_csv/transform.py:190]: what's with this `, `?

File: tap_s3_csv/transform.py [hunk 3]
```diff
@@ -216,8 +216,9 @@ def _transform_object(self, data, schema, path, pattern_properties):
                                if re.match(pattern, key)]
             if key in schema or pattern_schemas:
                 sub_schema = schema.get(key, {'anyOf': pattern_schemas})
+
                 success, subdata = self.transform_recur(
-                    value, sub_schema, path + [key])
+                    value, sub_schema, path + [key], source_type_map, source_type_map[key] if key in source_type_map else None)
```
Comment (kchan-varicent) [inline tap_s3_csv/transform.py:221]: `source_type_map.get(key)` does exactly that

File: tap_s3_csv/transform.py (hunk 1 above)
Comment (kchan-varicent) [reply tap_s3_csv/transform.py:392]: it's a dict isn't it?

**Review by kchan-varicent — COMMENTED** [review]
I thought we were going to write the source_type during discover and then on sync, use source_type only if the field is in columnsToUpdate

**Pull request closed without merging**